### ISSUE TITLE
Create Private ALB for Backend Service

### ISSUE THREAD

**Issue by flemzord** [aws/copilot-cli#2961]
Hello, 

It's possible to add Private ALB for Backend Service ? 
This would be much better for the scalability of a backend service.

Regards,
Maxence

**Comment by efekarakus** [comment]
Hello @flemzord 👋 

Thank you for the feature request! At the moment this feature isn't yet available for Backend Services.  
Out of curiosity, would you expect this feature to work just like Load Balanced Web Services? where only one ALB gets created per environment and it gets shared between services?

````
                                       ┌──────────────────────┐
                                       │   Backend Service    │
                          /svc1        │                      │
┌─────────────────────┐                │       "svc1"         │
│                     │ ──────────────►│                      │
│   Private ALB       │                └──────────────────────┘
│                     │
│                     │
│                     │  /svc2         ┌──────────────────────┐
│                     │ ──────────────►│                      │
└─────────────────────┘                │   Backend Service    │
                                       │                      │
                                       │       "svc2"         │
                                       └──────────────────────┘
````

**Comment by flemzord** [comment]
Exactly @efekarakus , the same as for the public part but in private :)

**Comment by jsutton** [comment]
I would find this useful. I can see it either being an option on "Load Balanced Web Service":
```
# Distribute traffic to your service.
http:
  type: internal
```
Or, adding an http section to "Backend Service".

It should still use ACM for certs, which I think means it would need to use a public Route53 zone.

**Comment by efekarakus** [comment]
Thanks for the feedback @jsutton !

If you don't mind, can you elaborate on the usecase for the internal ALB? is it also for better scaling properties?

> It should still use ACM for certs, which I think means it would need to use a public Route53 zone.

Would you expect TLS to be terminated at the ALB as well here? I'm not certain if a public route53 zone is needed, we'll test this out!

**Comment by flemzord** [comment]
The URL can be public, if the ALB is not public it doesn't create a real problem :) 

The metrics and scalability offered by an ALB are far superior to Cloud Map. 
For the creation of ACM certificates, a public zone is mandatory.

**Comment by jsutton** [comment]
The use case is internal services to be accessed over VPN or direct connect.

**Comment by flemzord** [comment]
Not only that @jsutton , in a Micro-Service strategy, but we may also want to use a Private ALB to put our Micro-Services that should not be accessible from the Internet.

**Comment by jsutton** [comment]
@flemzord for sure, I just don't have that use case in front of me at present so I can't answer questions about it as easily.

**Comment by FakieSui** [comment]
Interested aswell by this feature

**Comment by efekarakus** [comment]
Hi Antoine 😄 ! We're finalizing the design this week and should have a design proposal by end of week that we can post here.

**Comment by huanjani** [comment]
Q for those interested in this feature: 
Would you want
1. all resources within the VPC to have access to the internal load balancer, or
2. only Copilot services to have access?
Thanks!

**Comment by jsutton** [comment]
For me, all resources. I would still like the allowed IP list though.

**Comment by huanjani** [comment]
This feature is now released in v1.19.0: https://github.com/aws/copilot-cli/releases/tag/v1.19.0! 🎉
